### ISSUE TITLE
Make python's log-format editable by others using sanic

### ISSUE THREAD

**Issue by x10an14** [channelcat/sanic#176]
Hiya,

Sanic uses python's logging (in `sanic/log.py`) in a fashion which overrides the ability of others to set it they way they want to.

Could this be refactored into making it configurable by others? I'm not that into python logging/adept at it, but I'll try to make a PR this weekend if nobody else beat me to it!

Example code which gets overwritten in my script:
```python
(...)
from src import app # -> from sanic import Sanic; app = Sanic("my-module")
(...)

if __name__ == '__main__':
    debug = False

    logging_format = "[%(asctime)s] %(process)d-%(levelname)s "
    logging_format += "%(module)s::%(funcName)s():l%(lineno)d: "
    logging_format += "%(message)s"
    logging.basicConfig(
        format=logging_format,
        datefmt=default_parse_fmt + "%f",
        level=logging.DEBUG if debug else logging.INFO, )

    app.run(host="127.0.0.1", port=8000, debug=debug)
```

`sanic/log.py`'s config (below) overwrites mine, even though I only invoke it by importing sanic in my own `__init__.py` file:
```python
import logging

logging.basicConfig(
    level=logging.INFO, format="%(asctime)s: %(levelname)s: %(message)s")
log = logging.getLogger(__name__)
```

The `log.py` module in sanic is barely used anywhere. I easily managed to remove it and replace it with the use specified [here](https://docs.python.org/3.5/howto/logging.html#logging-from-multiple-modules). Everything still worked!

Proof:
```
[29-11-2016 20:04:07%f] 19034-DEBUG sanic::run():l281 >
                 ▄▄▄▄▄
        ▀▀▀██████▄▄▄       _______________
      ▄▄▄▄▄  █████████▄  /                 \
     ▀▀▀▀█████▌ ▀▐▄ ▀▐█ |   Gotta go fast!  |
   ▀▀█████▄▄ ▀██████▄██ | _________________/
   ▀▄▄▄▄▄  ▀▀█▄▀█════█▀ |/
        ▀▀▀▄  ▀▀███ ▀       ▄▄
     ▄███▀▀██▄████████▄ ▄▀▀▀▀▀▀█▌
   ██▀▄▄▄██▀▄███▀ ▀▀████      ▄██
▄▀▀▀▄██▄▀▀▌████▒▒▒▒▒▒███     ▌▄▄▀
▌    ▐▀████▐███▒▒▒▒▒▐██▌
▀▄▄▄▄▀   ▀▀████▒▒▒▒▄██▀
          ▀▀█████████▀
        ▄▄██▀██████▀█
      ▄██▀     ▀▀▀  █
     ▄█             ▐▌
 ▄▄▄▄█▌              ▀█▄▄▄▄▀▀▄
▌     ▐                ▀▀▄▄▄▀
 ▀▀▄▄▀

[29-11-2016 20:04:07%f] 19034-INFO sanic::run():l284 > Goin' Fast @ http://127.0.0.1:8000
[29-11-2016 20:04:11%f] 19034-INFO __init__::wrapper():l53 > Entering calculate_csv()...
[29-11-2016 20:04:11%f] 19034-DEBUG __init__::wrapper():l64 >   args={'request': <sanic.request.Request object at 0x7f17a6354bc8>}, kwargs={}
[29-11-2016 20:04:11%f] 19034-INFO webapp::calculate_csv():l56 > Receiving a request to /csv_upload
[29-11-2016 20:04:11%f] 19034-INFO webapp::calculate_csv():l59 > Received the following files:
[29-11-2016 20:04:11%f] 19034-INFO webapp::calculate_csv():l64 >        #of files: 1, file_names: [('csv_file', 'data.csv')]
[29-11-2016 20:04:11%f] 19034-INFO webapp::calculate_csv():l90 > Content writtent to tempfile...
[29-11-2016 20:04:11%f] 19034-INFO webapp::calculate_csv():l95 > CSV dialect '<class 'csv.Sniffer.sniff.<locals>.dialect'>' sniffed...
[29-11-2016 20:04:11%f] 19034-INFO webapp::calculate_csv():l100 > #rows in csv_file: 70
[29-11-2016 20:04:11%f] 19034-INFO webapp::calculate_csv():l102 > Parsing time records...
[29-11-2016 20:04:11%f] 19034-INFO __init__::wrapper():l53 > Entering parse_csv_reader_content()...
```

**Comment by wsbmilk** [comment]
debug=debug or True?

**Comment by wsbmilk** [comment]
as python official said:
the logging.basicConfig cannot be overrided.

**Comment by seemethere** [comment]
This feature will be rolled into 0.2.0

**Comment by seemethere** [comment]
Closed with #212
